cupyx/scipy/sparse/linalg/_solve.py

```diff
@@ -390,6 +392,24 @@ def lsmr(A, b, x0=None, damp=0.0, atol=1e-6, btol=1e-6, conlim=1e8,
     return x, istop, itn, normr, normar, normA, condA, normx
 
 
+def _does_use_spsm(b):
+    if not runtime.is_hip:
+        # cusparseSpSM in CUDA 11.5 and eariler produces wrong results with
```

Comment:
Let's use spsm starting CUDA 12.0 as this PR needs to be backported and impacts existing users.